openml-datarobot/src/main/java/com/feedzai/openml/datarobot/DataRobotModelCreator.java

```diff
@@ -77,6 +77,17 @@ public ClassificationBinaryDataRobotModel loadModel(final Path modelPath,
         ClassificationValidationUtils.validateParamsModelToLoad(this, modelPath, schema, ImmutableMap.of());
 
         final Pair<Predictor, URLClassLoader> predictorPair = createPredictorInstance(modelPath);
+        final int predictorSize = predictorPair.getKey().get_double_predictors().length + predictorPair.getKey().get_string_predictors().length;
+        // We ignore the target variable in the schema for schema matching purposes.
+        if (predictorSize != schema.getFieldSchemas().size() - 1) {
+            final String errorMsg = String.format(
+                    "Wrong number of fields in the given schema. The model expected %d but the schema had %d.",
```

Comment:
Or we could go one step further and show the diff between the predictor and the schema :D

Reply:
Good points guys, I'll log both schemas as it yields all the necessary info (maybe just printing the diff does not give the full picture clearly to the user).

I don't want to send it all in the error as this would be a huge message and the user interfaces won't deal very nicely with it.